pkg/util/docker/network.go

```diff
@@ -137,6 +137,60 @@ func resolveDockerNetworks(containerNetworks map[string][]dockerNetwork) {
 
 // DefaultGateway returns the default Docker gateway.
 func DefaultGateway() (net.IP, error) {
+	fields, err := defaultGatewayFields()
+	if err != nil || len(fields) < 3 {
+		return nil, err
+	}
+
+	ipInt, err := strconv.ParseUint(fields[2], 16, 32)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse ip %s from route file: %s", fields[2], err)
+	}
+	ip := make(net.IP, 4)
+	binary.LittleEndian.PutUint32(ip, uint32(ipInt))
+	return ip, nil
+}
+
+// DefaultHostIPs returns the IP addresses bound to the default network interface.
+// The default network interface is the one connected to the network gateway, and it is determined
+// by parsing the routing table file in the proc file system.
+func DefaultHostIPs() ([]string, error) {
+	fields, err := defaultGatewayFields()
+	if err != nil {
+		return nil, err
+	}
+	if len(fields) == 0 {
+		return nil, fmt.Errorf("missing interface information from routing file")
+	}
+	iface, err := net.InterfaceByName(fields[0])
+	if err != nil {
+		return nil, err
+	}
+
+	addrs, err := iface.Addrs()
+	if err != nil {
+		return nil, err
+	}
+
+	ips := make([]string, len(addrs))
+	for i, addr := range addrs {
+		// Translate CIDR blocks into IPs, if necessary
+		ips[i] = strings.Split(addr.String(), "/")[0]
```

Comment:
could do this without allocating using strings.IndexByte and slice offsets, but I don't think it matters in this context and this is more readable IMO.